src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                int amount = Integer.parseInt(parts[1]);
+
+                if ("supply".equals(parts[0])) {
+                    totalSupply += amount;
+                } else if ("buy".equals(parts[0])) {
+                    totalBuy += amount;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading from file: " + fromFileName, e);
+        }
+
+        int result = totalSupply - totalBuy;
 
+        try (FileWriter writer = new FileWriter(toFileName)) {
+            writer.write("supply," + totalSupply + System.lineSeparator());
+            writer.write("buy," + totalBuy + System.lineSeparator());
+            writer.write("result," + result + System.lineSeparator());
```

Comment:
You are using string concatenation inside a loop. According to the checklist, you should use `StringBuilder` instead of `String` to concat data in loops. Keep in mind that String concatenation creates many new objects that take up a lot of memory if you use it inside of a loop. Though it's safe to use it outside of a loop because the compiler will replace it with StringBuilder anyway [java doc](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html).